controllers/noderefutil/providerid.go

```diff
@@ -89,7 +88,7 @@ func (p *ProviderID) ID() string {
 
 // Equals returns true if both the CloudProvider and ID match.
```

Comment:
comment should be updated to match new function (or we should make a new func as @killianmuldoon suggests)

Reply:
How might we deprecate this in favor of another func and not affect users? AFAICT this func is only used here:

https://github.com/kubernetes-sigs/cluster-api/blob/release-1.1/internal/controllers/machine/machine_controller_noderef.go#L196

If we created a new func and updated the reference in `(r *Reconciler) getNode` we would still be making a change for all providers, no?

Reply:
I think we need to deprecate it and replace as it's a part of our public Go API, so we don't have control over who is using it and for what.

As for the behavioural change inside the CAPI method, I think the impact of that needs to be assessed separately. If there is an impact from changing the behaviour here that should be called out in the deprecation notice and/or on the migration guide.

Reply:
@jackfrancis here's an example of deprecating a public func in CAPI https://github.com/kubernetes-sigs/cluster-api/pull/5545/files#diff-e25fb86695a8022fa1611100bd244f9b4185f28629d19830e83d18b4fa2ce710R41

And an example of adding to the migration guide for providers https://github.com/kubernetes-sigs/cluster-api/commit/f73a27729acf3bc2cce2479220e1460eb6f73efb#diff-531b360cba021aa4e0cf2df8f2ce8ead9601db1601bf516cab3c916a83622436R85